### ISSUE TITLE
Feature-request: Authorization by tokens for /metrics

### ISSUE THREAD

**Issue by kvaps** [LINBIT/linstor-server#166]
Hi, currently `/metrics` can be exported only via same connector (plain or SSL), which used for accessing the LINSTOR API itself.

Prometheus [does support](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#tls_config) mutual SSL authentication for scraping targets:

```yaml
- job_name: 'node_exporter_ssl'
    scrape_interval: 5s
    scheme: https
    tls_config:
      ca_file: /etc/ssl/certs/prometheus-ca.crt
      cert_file: /etc/ssl/certs/prometheus.crt
      key_file: /etc/prometheus/prometheus.key
    static_configs:
      - targets: ['${TARGET_IP}:443']
```
But it is not so common.

Please correct me if I'm wrong, but the problem is also compounded by the fact that I can't give an access to `/metrics` without providing access for managing whole cluster.

Usually `/metrics` are accessible without authentication, with base HTTP authentification or using special access token.

I like the last option, for example Gitlab is using it. Before configuring targets you have to generate an access token:
![Screenshot_2020-08-10 Health Check · Admin Area](https://user-images.githubusercontent.com/7556217/89773447-32fcc580-db04-11ea-921a-6ada5b8867c1.png)

And after that your metrics can be simple accessible on:

```
https://gitlab.example.org/-/metrics?token=thiepuleay2ColaM7wai
```

I'm sure such method is fair enough to provide simple and secure way for collecting metrics.
Also it could easily be implemented in the linstor application logic.

## Interface changes

This is my vision of interface changes for this:
```bash
linstor controller set-property netcom/Metrics/token Aht7hie0ohjeyuT0ohju
```

and after that metrics should be accessible on:
```bash
curl -k 'https://localhost:3370/metrics?token=Aht7hie0ohjeyuT0ohju'
```

Under https scheme but without client certificate authentication.

Optionally if token is not set, then `/metrics` can be accessed the same way, but without an authentication.

**Comment by rp-** [comment]
I don't think I can remove mutual SSL authentication just for the `/metrics` url, as the SSL configuration is global to all paths on the SSL port.
But I can anyway add this token authorization and do not redirect on HTTP requests, so anyone can use the HTTP port with the correct token.
But than I might still need some option for super paranoid users who don't want people have access anything on a plain HTTP port?

**Comment by kvaps** [comment]
Token authentication on plain HTTP make no much sense anyone can sniff the traffic and get the token.

If `/metrics` path is secured by SSL, you can always enable strict mode and make sure that connection will be interrupted before sending the token if certificate for the controller is not valid.

In this way I would prefer to have no any authentication for `/metrics`, but have an opportunity to enable it by using external tools, eg [kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy) to add an authentication if it is really needed.

**Comment by kvaps** [comment]
I think we just need simple option to allow access on `/metrics` via plain http listener

**Comment by rp-** [comment]
do you just mean, that `/metrics` also works if HTTPS is enabled?

**Comment by kvaps** [comment]
I want an option to allow `/metrics` on simple HTTP.
Currently they are accessible only via HTTPS with mutual authentication.

But I think it should be some configuration option, because sometime you might want to restrict an access to `/metrics`

**Comment by rp-** [comment]
got it

**Comment by rp-** [comment]
should be fixed in the final 1.8.0.
there is now a controller property `REST/disable-http-metrics` to disable `/metrics` url for HTTP
